components/tidb_query/src/codec/mysql/binary_literal.rs

```diff
@@ -0,0 +1,468 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::cmp::{Eq, Ord, Ordering, PartialEq, PartialOrd};
+use std::string::ToString;
+
+use byteorder::{BigEndian, WriteBytesExt};
+use hex;
+
+use crate::codec::error::Error;
+use crate::codec::Result;
+use crate::expr::EvalContext;
+
+/// BinaryLiteral is the internal type for storing bit / hex literal type.
+#[derive(Debug)]
+pub struct BinaryLiteral(Vec<u8>);
+
+fn trim_leading_zero_bytes(bytes: &[u8]) -> &[u8] {
```

Comment:
https://doc.rust-lang.org/std/primitive.str.html#method.trim_start_matches

Reply:
This method cannot be applied for `&[u8]`.